### DISCUSSION TITLE
How to initialize C string directly?

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1082] [General]
From the document it seems you can't do sth like this:

var s: Cstring.char_star = "text";

https://modula3.github.io/cm3/help/gen_html/m3core/src/C/FreeBSD3/Cstring.i3.html

https://modula3.github.io/cm3/help/gen_html/m3core/src/C/Common/M3toC.i3.html

You have to create a variable with type TEXT, initialize it: var t: TEXT = "text"; then use M3toC's functions to copy the value of it to s: s := M3toC.CopyTtoS(t);

I wanted to know if there is a way to create and initialize the C string directly. Thanks.

**Comment by mikanystrom** [comment]
maybe you can get away with

VAR s := ARRAY [0..4] OF CHAR { 't', 'e', 'x', 't', '\000' };

or somesuch thing.  Then pass ADR(s) to your C function.  (Careful about
going out of scope, though.)

**Comment by RodneyBates** [comment]
On 9/14/22 07:08, zpu2tw wrote:
>
> From the document it seems you can't do sth like this:
>
> var s: Cstring.char_star = "text";
>
> https://modula3.github.io/cm3/help/gen_html/m3core/src/C/FreeBSD3/Cstring.i3.html
>
> https://modula3.github.io/cm3/help/gen_html/m3core/src/C/Common/M3toC.i3.html
>
> You have to create a variable with type TEXT, initialize it: var t: TEXT = "text"; then use M3toC's functions to copy the value of it to s: s := M3toC.CopyTtoS(t);
>
> I wanted to know if there is a way to create and initialize the C string directly. Thanks.
>

VAR s: Cstring.char_star = M3toC.CopyTtoS("text")

But you when there are two different representations of
strings, you could not avoid coding something (CopyTtoS)
to denote which one.
